quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/QuarkusCli38to315UpdateIT.java

```diff
@@ -0,0 +1,451 @@
+        Properties oldProperties = new Properties();
+        oldProperties.put("quarkus.package.type", "native-sources");
+
+        Properties expectedNewProperties = new Properties();
+        expectedNewProperties.put("quarkus.package.jar.enabled", "false");
+        expectedNewProperties.put("quarkus.native.enabled", "true");
+        expectedNewProperties.put("quarkus.native.sources-only", "true");
+
+        QuarkusCLIUtils.checkPropertiesUpdate(quarkusCLIAppManager, oldProperties, expectedNewProperties);
+    }
+
+    @Test
+    public void updateResteasyReactiveAppTest() throws XmlPullParserException, IOException {
+        QuarkusCliRestService app = cliClient.createApplicationFromExistingSources("app", null, RESTEASY_REACTIVE_APP);
+        QuarkusCLIUtils.setCommunityBomIfNotRunningRHBQ(app, OLD_COMMUNITY_VERSION);
+
+        quarkusCLIAppManager.updateApp(app);
+
+        List<Dependency> dependencies = getDependencies(app);
+        assertFalse(dependencies.contains(new QuarkusDependency("io.quarkus:quarkus-resteasy-reactive")),
+                "quarkus-resteasy-reactive should be renamed to quarkus-rest");
+        assertFalse(dependencies.contains(new QuarkusDependency("io.quarkus:quarkus-rest-client-reactive-jackson")),
+                "quarkus-rest-client-reactive-jackson should be renamed to quarkus-rest-client-jackson");
+
+        assertTrue(dependencies.contains(new QuarkusDependency("io.quarkus:quarkus-rest")),
+                "quarkus-resteasy-reactive should be renamed to quarkus-rest");
+        assertTrue(dependencies.contains(new QuarkusDependency("io.quarkus:quarkus-rest-client-jackson")),
+                "quarkus-rest-client-reactive-jackson should be renamed to quarkus-rest-client-jackson");
+
+        app.start();
+        app.given().get("/book").then().statusCode(HttpStatus.SC_OK);
+    }
+
+    @Test
+    public void updateMessagingAmqpAppTest() throws XmlPullParserException, IOException {
+        QuarkusCliRestService app = cliClient.createApplicationFromExistingSources("app", null, MESSAGING_APP);
+        QuarkusCLIUtils.setCommunityBomIfNotRunningRHBQ(app, OLD_COMMUNITY_VERSION);
+
+        quarkusCLIAppManager.updateApp(app);
+
+        List<Dependency> dependencies = getDependencies(app);
+        assertFalse(dependencies.contains(new QuarkusDependency("io.quarkus:quarkus-smallrye-reactive-messaging-amqp")),
+                "quarkus-smallrye-reactive-messaging-amqp should be renamed to quarkus-rest");
+
+        assertTrue(dependencies.contains(new QuarkusDependency("io.quarkus:quarkus-messaging-amqp")),
+                "quarkus-messaging-amqp should be renamed to quarkus-rest");
+
+        app.start();
+        assertTrue(app.given().get("/price").thenReturn().body().asString().contains("20"),
+                "App should return not-empty result");
+    }
+
+    @Test
+    public void updateHibernateAppTest() throws XmlPullParserException, IOException {
```

Comment:
Please add comment what is the purpose of this test, from what migration guide it comes

Reply:
comment added.